platform/android/audio_driver_opensl.cpp

```diff
@@ -330,3 +365,7 @@ void AudioDriverOpenSL::set_pause(bool p_pause) {
 
 AudioDriverOpenSL::AudioDriverOpenSL() {
 }
+
+AudioDriverOpenSL::~AudioDriverOpenSL() {
+	end();
```

Comment:
It's unnecessary, don't think the destructor of `AudioDriverOpenSL` can be called without `finish` if everything initialized properly. Here is the order of calls as far as I'm aware: [`_terminate()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/platform/android/java_godot_lib_jni.cpp#L78) -> [`Main::cleanup()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/platform/android/java_godot_lib_jni.cpp#L96) -> [`Main::cleanup()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/main/main.cpp#L4220) (copied to include another link) -> [`audio_server->finish()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/main/main.cpp#L4309) -> [`AudioServer::finish()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/servers/audio_server.cpp#L1494) -> [`AudioDriver::finish()`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/servers/audio_server.cpp#L1496) -> [`delete os_android`](https://github.com/godotengine/godot/blob/292e50e17e3f6e2509d3178a00204f964a907460/platform/android/java_godot_lib_jni.cpp#L97) -> `~OS_Android()` -> `~AudioDriverOpenSL()`. If it crashes, no destructors are called. If `AudioServer` was not created, `AudioDriverOpenSL::init`, `AudioDriverOpenSL::start` and `AudioDriverOpenSL::input_start` can't be called

Reply:
I don't see any harm in leaving this here since if `end()` has already been called, the second call will be a no-op.

Reply:
There's no harm, but it's not needed too. It also creates inconsistency with other `AudioDriver`s as no other `AudioDriver` defines a deconstructor, there's already a function `finish` for this. It's more of a style change than anything

Reply:
I did a quick test and the fix still works as expected with the destructor removed so it can be removed for consistency.

Reply:
I pushed an updated removing that destructor.

I also removed the explicit `~OS_Unix` destructor and `virtual` qualifier for `~OS_Android` which were discussed above. They're not wrong, but also don't seem needed for that fix, and would introduce an inconsistency between `OS_Android` and other OS implementations.

If we want to make sure all classes that inherit one with a `virtual` destructor also make it explicit in their declaration, that should likely be a separate style fix (as I understand it it shouldn't impact the behavior: https://stackoverflow.com/questions/71249522/virtual-destructor-needed-for-class-which-is-both-derived-and-base).

Still would be worth testing the latest commit again to make sure.

Reply:
Checked the latest push and `libOpenSLES: Object::Destroy` errors are gone